pandas/io/parsers.py

```diff
@@ -2289,7 +2289,11 @@ def __init__(self, f: Union[FilePathOrBuffer, List], **kwds):
             self._open_handles(f, kwds)
             assert self.handles is not None
             assert hasattr(self.handles.handle, "readline")
-            self._make_reader(self.handles.handle)
+            try:
+                self._make_reader(self.handles.handle)
+            except:
+                # self.close()
```

Comment:
why don't you call `self.close()` but put a try except around the actual os.remove call (and ignore an OS exception)

Reply:
I want to make sure that the test actually works - at the moment it doesn't. Once I get it working, I will use the `self.close()` call. The user (not pandas) is calling `os.remov`. I think we should still make sure that we do not have `ResourceWarning`s, that is what I'm trying to fix.